configure.ac

```diff
@@ -88,6 +88,11 @@ AC_FUNC_STRERROR_R
 
 AC_PATH_TOOL([PKG_CONFIG], [pkg-config], [pkg-config])
 
+#
+# Define SHARED_LIB preprocessor macro when building a shared library
+#
+lt_prog_compiler_pic="$lt_prog_compiler_pic -DUCS_SHARED_LIB"
```

Comment:
not sure UCS prefix is good here since it affects all files actually

Reply:
> not sure UCS prefix is good here since it affects all files actually

renamed to UCX_SHARED_LIB... just SHARED_LIB is too general name, could be conflicts